consensus/service/src/api/client_api_service.rs

```diff
@@ -82,7 +82,19 @@ impl ClientApiService {
         msg: Message,
     ) -> Result<ProposeTxResponse, ConsensusGrpcError> {
         counters::ADD_TX_INITIATED.inc();
-        let tx_context = self.enclave.client_tx_propose(msg.into())?;
+        let tx_context = match self.enclave.client_tx_propose(msg.into()) {
+            Ok(tx_context) => tx_context,
+            Err(EnclaveError::FeeMapDigestMismatch) => {
+                let mut response = ProposeTxResponse::new();
+                response.set_result(ProposeTxResult::FeeMapDigestMismatch);
+                response.set_err_msg(EnclaveError::FeeMapDigestMismatch.to_string());
+                return Ok(response);
+            }
```

Comment:
this part is confusing to me, I think `err.into()` should lead to the same result?

Reply:
can we do this special casing at e.g. line 57 in `consensus/service/src/api/grpc_error.rs` instead?

```
impl From<EnclaveError> for ConsensusGrpcError {
    fn from(src: EnclaveError) -> Self {
        match src {
            EnclaveError::MalformedTx(err) => Self::from(err),
            _ => Self::Enclave(src),
        }
    }
}
```

Reply:
I see, that's not going to set the `err_msg` field. But I'm not sure thats a big deal, the client only reads that in case of a TransactionValidationError I think

Reply:
It would be nice if we could figure out how to keep plumbing like this in the `conversions.rs` module as opposed to here in the business logic

Reply:
Lol, this was copied from https://github.com/mobilecoinfoundation/mobilecoin/pull/2670 :)
I'll take a look and see how to relocate it.